src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -712,6 +970,41 @@ private final class BitmapIndexedMapNode[K, +V](
     )
   }
 
+  /** Variant of `copyAndMigrateFromNodeToInline` which mutates `this` rather than returning a new node.
+    *
+    * @param bitpos the bit position of the node to migrate inline
+    * @param oldNode the node currently stored at position `bitpos`
+    * @param node the node containing the single key-value pair to migrate inline
+    */
+  def migrateFromNodeToInlineInPlace[V1 >: V](bitpos: Int, oldNode: MapNode[K, V1], node: MapNode[K, V1]): Unit = {
+    val idxOld = this.content.length - 1 - nodeIndex(bitpos)
+    val dataIxNew = dataIndex(bitpos)
+    val idxNew = TupleLength * dataIxNew
+
+    val key = node.getKey(0)
+    val value = node.getValue(0)
+    val src = this.content
+    val dst = new Array[Any](src.length - 1 + TupleLength)
```

Comment:
It's a shame that we have to allocate a whole new array just to store one more element.  I guess that's unavoidable?

Reply:
Probably unavoidable ere, but we may be able to do something about this in the equivalent HashSet code, since TupleLength == 1 there.